### PR TITLE
Message max length

### PR DESCRIPTION
**Pull request by janbarasek**
- new feature
- BC break? yes

In case of extremely length error message BlueScreen log full message. I think it should be truncated for example to `32768 characters`.

Example code: `throw new \Exception(str_repeat('abcd ', 10000));`.

Snímek obrazovky 2019-09-11 v 12 50 23

### CODE REVIEW

File: src/Tracy/Helpers.php [hunk 1]
```diff
@@ -157,6 +157,43 @@ public static function errorTypeToString(int $type): string
 	}
 
 
+	/**
+	 * Truncates UTF-8 string to maximal length.
+	 */
+	public static function truncate(string $s, int $maxLen = 32768, string $append = "\u{2026}"): string
```
Comment (JanTvrdik) [inline src/Tracy/Helpers.php]: This is too complex for Tracy. You should extract the truncate code from Dumper

Comment (JanTvrdik) [reply src/Tracy/Helpers.php]: https://github.com/nette/tracy/blob/44349212f8856cf1cacb0ab2b62c2b30e4be6f93/src/Tracy/Dumper/Dumper.php#L519-L531

Comment (dg) [reply src/Tracy/Helpers.php]: I added method Dumper::truncateString()

Comment (janbarasek) [reply src/Tracy/Helpers.php]: @dg Thank you and replaced `Helpers::truncate` to `Dumper::encodeString`.

**Comment by dg** [comment]
thanks

**Pull request merged**